tests/unit/discovery/windows/conftest.py

```diff
@@ -0,0 +1,169 @@
+            raise result
+        return result
+
+    mocker.patch.object(winreg, "QueryValueEx", side_effect=_v)
+
+    class Key:
+        def __init__(self, value):
+            self.value = value
+
+        def __enter__(self):
+            return self
+
+        def __exit__(self, exc_type, exc_val, exc_tb):  # noqa: U100
+            return None
+
+    @contextmanager
+    def _o(*args):
+        if len(args) == 2:
+            key, value = args
+            key_id = key.value if isinstance(key, Key) else key
+            result = Key(key_open[key_id][value])  # this needs to be something that can be with-ed, so let's wrap it
+        elif len(args) == 4:
+            result = hive_open[args]
+        else:
+            raise RuntimeError
+        value = result.value if isinstance(result, Key) else result
+        if isinstance(value, OSError):
+            raise value
+        yield result
+
+    mocker.patch.object(winreg, "OpenKeyEx", side_effect=_o)
+    mocker.patch("os.path.exists", return_value=True)
+
+
+@pytest.fixture()
+def _collect_winreg_access(mocker):
+    # noinspection PyUnresolvedReferences
+    from winreg import EnumKey, OpenKeyEx, QueryValueEx
+
+    from virtualenv.discovery.windows.pep514 import winreg
+
+    hive_open = {}
+    key_open = defaultdict(dict)
+
+    @contextmanager
+    def _c(*args):
+        res = None
+        key_id = id(args[0]) if len(args) == 2 else None
+        try:
+            with OpenKeyEx(*args) as c:
+                res = id(c)
+                yield c
+        except Exception as exception:
+            res = exception
+            raise exception
+        finally:
+            if len(args) == 4:
+                hive_open[args] = res
+            elif len(args) == 2:
+                key_open[key_id][args[1]] = res
+
+    enum_collect = defaultdict(list)
+
+    def _e(key, at):
+        result = None
+        key_id = id(key)
+        try:
+            result = EnumKey(key, at)
+            return result
+        except Exception as exception:
+            result = exception
+            raise result
+        finally:
+            enum_collect[key_id].append(result)
+
+    value_collect = defaultdict(dict)
+
+    def _v(key, value_name):
+        result = None
+        key_id = id(key)
+        try:
+            result = QueryValueEx(key, value_name)
+            return result
+        except Exception as exception:
+            result = exception
+            raise result
+        finally:
+            value_collect[key_id][value_name] = result
+
+    mocker.patch.object(winreg, "EnumKey", side_effect=_e)
+    mocker.patch.object(winreg, "QueryValueEx", side_effect=_v)
+    mocker.patch.object(winreg, "OpenKeyEx", side_effect=_c)
+
+    yield
+
+    print("")
+    print(f"hive_open = {hive_open}")
+    print(f"key_open = {dict(key_open.items())}")
+    print(f"value_collect = {dict(value_collect.items())}")
+    print(f"enum_collect = {dict(enum_collect.items())}")
```

Comment:
Remove prints please, I think were forgotten here 🤔

Reply:
This entire fixture function is unused, as far as I can see. Function removed.